content/client-lib-development-guide/features.textile

```diff
@@ -696,3 +701,355 @@ The following default values are configured for the client library:
 * @(DF1)@ Realtime defaults:
 ** @(DF1a)@ @connectionStateTtl@ integer - default 60s. The duration that Ably will persist the connection state when a Realtime client is abruptly disconnected. When the client is in the @DISCONNECTED@ state, once this TTL has passed, the client should change the state to the @SUSPENDED@ state signifying that the state is now lost i.e. channels need to be reattached manually. Note that this default is overriden by @connectionStateTtl@, if specified in the @ConnectionDetails@ of the @CONNECTED@ @ProtocolMessage@
 ** @(DF1b)@ @realtimeRequestTimeout@ - default 10s. When a realtime client library is establishing a connection with Ably, or sending a @HEARTBEAT@, @CONNECT@, @ATTACH@, @DETACH@ or @CLOSE@ @ProtocolMessage@ to Ably, this is the amount of time that the client library will wait before considering that request as failed and triggering a suitable failure condition
+
+h2(#idl). Interface Definition
+
+The following bespoke IDL (Interface Definition Language) describes the types and classes present in the Rest and Realtime client libraries.
+
+Please note the following conventions:
+
+* Types are capitalized
+* Variable and method names are lowercase
+* Values produced by I/O (e. g. a request) are prefixed with @io@. In some platforms (JS) those values are passed as arguments to a callback argument, instead of being returned. I/O always can fail, but how do they fail is undefined in the spec, so it's also undefined here
+* Functions are denoted as @(Type, argName: Type, argName: Type) -> ReturnType@ (@ReturnType@ may be omitted)
+* Enums are denoted as @.A | .B | .C@
+* @Type?@ denotes an optional type, ie. a type that can be null plus all values of type Type
+* @Type default value@ denotes that the thing being anotated with @Type@ has @value@ as default. @Type api-default value@ denotes that the Ably server API uses those defaults and therefore it is unnecessary for the client library to send these values by default
+* Class fields (as opposed to instance fields) are prefixed with a @+@
+
+```[python]
+class Rest:
+  constructor: (keyStr: String) -> // RSC1
+  constructor: (tokenStr: String) -> // RSC1
+  constructor: (ClientOptions) -> // RSC1
+  auth: Auth // RSC5
+  channels: Channels<Channel> // RSN1
+  stats: (
+    start: Time, // RSC6b1
+    end: Time api-default now(), // RSC6b1
+    direction: .Backwards | .Forwards api-default .Backwards, // RSC6b2
+    limit: int api-default 100, // RSC6b3
+    unit: .Minute | .Hour | .Day | .Month api-default .Minute // RSC6b4
+  ) -> io PaginatedResult<Stats> // RSC6a
+  time: () -> io Time // RSC16
+
+class Realtime:
+  constructor: (keyStr: String) -> // RSC1
+  constructor: (tokenStr: String) -> // RSC1
+  constructor: (ClientOptions) -> // RSC1
+  close: () // proxy for RTN12
```

Comment:
Ah I actually misunderstood the syntax, I thought anything with `functionName: ()` indicated a function, whereas without the `()` it was an attribute.  I assumed (incorrectly) that the arrow indicated a closure / callback of some sort so for example, if `close` took a callback, then it would be shown as `close(() ->)`.  Are you saying every function with this syntax should have a trailing `->`?  If it then wanted to express close with a callback would it be `close(() ->) ->`?